### PR TITLE
Add Headword Extractor Implementation

### PR DESCRIPTION
**Pull request by sbhaktha**
@schmmd : here it is

@aimichal : Schmmd suggested I include you on this.
I need a headword extractor for something I am building in the DA solver. I know at least one other person who has needed this in the past. We had a `trait` for a `HeadExtractor` defined in `nlpstack` but no implementation. This PR adds the [`UWHeadExtractor` implementation from the knowitall project](https://github.com/knowitall/nlptools/blob/cbc2e3c518478358e6ca007440d2157a418f99ee/headword/uw/src/main/scala/edu/knowitall/tool/headword/UWHeadExtractor.scala).

This adds the core implementation from the `nlptools`project as is, with appropriate plumbing, and in addition contains the following things-
* formatting fixes
* couple of new convenience functions
* basic unit tests

### CODE REVIEW

**Comment by schmmd** [comment]
LGTM!

**Comment by dirkgr** [comment]
Woha! Nlpstack lives!

File: tools/headword/src/test/scala/KnowitallHeadExtractorSpec.scala [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  "KnowitallHeadExtractor" should
+    "return the right head word for noun phrases" in {
+      assert(
+        (testNp1Result === testNp1ExpectedHeadStr) &&
+          (testNp2Result === testNp2ExpectedHeadStr) &&
+          (testNp3Result === testNp3ExpectedHeadStr) &&
+          (testNp4Result === testNp4ExpectedHeadStr) &&
+          (testNp5Result === testNp5ExpectedHeadStr)
+      )
+    }
```
Comment (aimichal) [inline tools/headword/src/test/scala/KnowitallHeadExtractorSpec.scala]: Consider factoring this into one test per input/output pair. That would make the output more granular, and then this code would reads a little more clearly IMHO.

For example, you could make a helper and then write five tests, one of which might execute `testNounPhrase("walking quickly", "walking")`.

Comment (schmmd) [inline tools/headword/src/test/scala/KnowitallHeadExtractorSpec.scala]: Everything that isn't Sumithra's is probably my old crappy code ;-)

**Comment by sbhaktha** [comment]
@schmmd , @aimichal  : Finally Semaphore passed! Semaphore's support team got back to me and I was able to use their tips to access the logs. The failed builds were because `nlpstack` was still pointing to the Nexus Sonatype repo which we have deprecated. Switching to the new `allenai` sbt plugins to use the right resolver(s) led to many cascading dependency errors. If you're interested in looking at the changes, please take a look at the latest commit, if not I'll just merge. They're all just `build.sbt` changes.
@schmmd : it would be good to explicitly assign owners (teams) to every repo. This one was forgotten during the transition!

**Comment by schmmd** [comment]
> This one was forgotten during the transition!

@sbhaktha how would owners have prevented this?  How would the owner know the check out and evaluate this repository when we were deprecating Nexus?

**Comment by sbhaktha** [comment]
@schmmd : I was thinking of this email from you: https://groups.google.com/a/allenai.org/d/topic/team/EjmMNPKCqeI/discussion about the Nexus deprecation. The problem was not that the deprecation was not communicated but that there was no clear owner for this repo. Also it's not entirely clear from that thread whether `aristo` refers to just the repo or the team, so we could (a) assign owners for every repo (b) communicate, as you did, when there's a breaking change, so owners do the needful for their repos.

**Comment by aimichal** [comment]
@schmmd  wrote:

> @sbhaktha how would owners have prevented this? How would the owner know the check out and evaluate this repository when we were deprecating Nexus?

That requires a definition of "owner". IMHO, a repo owner is someone/someteam who cares about a repo's health (e.g., code can be compiled) and the resulting products health (e.g., if it's code for a website, then the website is up.) In the case of nexus deprecation, such a person/team would be sensitive to company-wide announcements about changes wrt their repo/product, and confirm that things won't be broken.

**Pull request merged**